server/db/042-rename-buttons-to-devices.sql

```diff
@@ -0,0 +1,91 @@
+        ALTER TABLE buttons_vitals DROP CONSTRAINT buttons_vitals_button_id_fkey;
+        ALTER TABLE buttons_vitals RENAME COLUMN button_id TO device_id;
+        ALTER TABLE buttons_vitals ADD CONSTRAINT buttons_vitals_device_id_fkey FOREIGN KEY (device_id) REFERENCES devices(id);
+        -- Enable the create_buttons_vitals_trigger trigger
+        ALTER TABLE buttons_vitals ENABLE TRIGGER create_buttons_vitals_trigger;
+        -- Replace the trigger to insert or update buttons_vitals_cache every time a new row is added to buttons_vitals
+        CREATE OR REPLACE FUNCTION create_buttons_vitals_trigger_fn()
+        RETURNS TRIGGER LANGUAGE PLPGSQL AS $t$
+        BEGIN
+            INSERT INTO buttons_vitals_cache (id, device_id, battery_level, created_at) 
+            VALUES (NEW.id, NEW.device_id, NEW.battery_level, NEW.created_at)
+            ON CONFLICT (device_id)
+            DO UPDATE SET
+                id = NEW.id,
+                battery_level = NEW.battery_level,
+                created_at = NEW.created_at;
+            RETURN NEW;
+        END;
+	$t$;
+
+        -- buttons_vitals_cache table changes
+
+        -- Disable the set_buttons_vitals_cache_timestamp trigger
+        ALTER TABLE buttons_vitals_cache DISABLE TRIGGER set_buttons_vitals_cache_timestamp;
+        -- Rename button_id column to device_id, while maintaining foreign-key constraint for id column in devices
+        ALTER TABLE buttons_vitals_cache DROP CONSTRAINT buttons_vitals_cache_button_id_fkey;
+        ALTER TABLE buttons_vitals_cache RENAME COLUMN button_id TO device_id;
+        ALTER TABLE buttons_vitals_cache ADD CONSTRAINT buttons_vitals_cache_device_id_fkey FOREIGN KEY (device_id) REFERENCES devices(id);
+        -- Enable the set_buttons_vitals_cache_timestamp trigger
+        ALTER TABLE buttons_vitals_cache ENABLE TRIGGER set_buttons_vitals_cache_timestamp;
+        
+        -- sessions table table changes
+
+        -- Disable the set_sessions_timestamp trigger
+        ALTER TABLE sessions DISABLE TRIGGER set_sessions_timestamp;
+        -- Add column is_resettable to table sessions, not null, defaulting to false
+        ALTER TABLE sessions ADD COLUMN is_resettable BOOLEAN NOT NULL DEFAULT 'f';
+        -- Rename button_id column to device_id, while maintaining foreign-key constraint for id column in devices
+        ALTER TABLE sessions DROP CONSTRAINT sessions_button_id_fkey;
+        ALTER TABLE sessions RENAME COLUMN button_id TO device_id;
+        ALTER TABLE sessions ADD CONSTRAINT sessions_device_id_fkey FOREIGN KEY (device_id) REFERENCES devices(id);
+        -- Enable the set_sessions_timestamp trigger
+        ALTER TABLE sessions ENABLE TRIGGER set_sessions_timestamp;
+
+        -- notifications table changes
+
+        -- Drop the notifications table if it exists, which is no longer used
+        DROP TABLE IF EXISTS notifications;
+
```

Comment:
Oo! I think that's a really cool idea, but would be very interesting to implement... I think I'll create a task for this and leave thinking about it for the future.

> For now, I suggest that for clients who have both Buttons and Sensors that instead of merging them into a single client, we leave them as two different ones. This will be a little confusing at first, but as long as Johnny knows what's going on, he should be able to tell which one is which pretty easily by looking at the incident categories.

Yes, I was thinking this too, I don't think I'll be merging clients together. Once a devices API is created, it should be easy to move all devices to a single "unified" clients for where a client has both buttons and sensor, and that client can be configured with the preferred incident categories, etc.